### DISCUSSION TITLE
Setting STORAGE_DRIVER to s3 doesn't appear to do anything

### DISCUSSION THREAD

**Discussion by jcsumlin** [docmost/docmost#439] [Help / Questions]
I recently stood up a docmost instance, and I'm really impressed with the project so far! The one feature I was looking forward to using was the s3 object storage for backing up my wiki. I am using the docker-compose file outlined by the project and while I think I've configured it all properly it doesn't seem to be doing anything since I switched it over to using "s3".

Here is my compose file:
```
version: '3'

services:
  docmost:
    image: docmost/docmost:latest
    depends_on:
      - db
      - redis
    environment:
      APP_URL: 'http://docmost.dataden.xyz'
      APP_SECRET: 'REDACTED'
      DATABASE_URL: 'postgresql://docmost:REDACTED@db:5432/docmost?schema=public'
      REDIS_URL: 'redis://redis:6379'
      STORAGE_DRIVER: 's3'
      AWS_S3_ACCESS_KEY_ID: 'B2_KEY_ID_REDACTED'
      AWS_S3_SECRET_ACCESS_KEY: 'B2_APPLICATION_KEY_REDACTED'
      AWS_S3_REGION: 'us-east-005'
      AWS_S3_BUCKET: 'docmost-s3'
      AWS_S3_ENDPOINT: 's3.us-east-005.backblazeb2.com'
    ports:
      - "3000:3000"
    restart: unless-stopped
    volumes:
      - docmost:/app/data/storage

  db:
    image: postgres:16-alpine
    environment:
      POSTGRES_DB: docmost
      POSTGRES_USER: docmost
      POSTGRES_PASSWORD: REDACTED
    restart: unless-stopped
    volumes:
      - db_data:/var/lib/postgresql/data

  redis:
    image: redis:7.2-alpine
    restart: unless-stopped
    volumes:
      - redis_data:/data

volumes:
  docmost:
  db_data:
  redis_data:
```

Am I using this wrong? The logs I'm seeing from the docmost container don't seem to communicate any failures.
```
> docmost@0.4.1 start /app
> pnpm --filter ./apps/server run start:prod


> server@0.4.1 start:prod /app/apps/server
> cross-env NODE_ENV=production node dist/main

(node:40) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[Nest] 40  - 10/25/2024, 7:42:27 PM     LOG [NestFactory] Starting Nest application...
[Nest] 40  - 10/25/2024, 7:42:28 PM     LOG [DatabaseModule] Establishing database connection
[Nest] 40  - 10/25/2024, 7:42:28 PM     LOG [DatabaseModule] Database connection successful
[Nest] 40  - 10/25/2024, 7:42:28 PM     LOG [DatabaseMigrationService] No pending database migrations
[Nest] 40  - 10/25/2024, 7:42:28 PM     LOG [NestApplication] Nest application successfully started
```

**Comment by Philipinho** [comment]
S3 is for file storage and not backups.
When you attach images and files to your pages, they will be stored in S3.

**Reply by jcsumlin** [reply]
Thanks, I assumed the storage driver was referring to the pages I made as well, not just the images.

I have tried making a few changes to my docker compose file to get this to work, but every time I try and upload a new image to a page, I get an `Invalid URL` error. I looked up the [b2 docs](https://www.backblaze.com/docs/cloud-storage-use-the-aws-sdk-for-javascript-v3-with-backblaze-b2) for the JavaScript SDK you're using, and I'm pretty sure I'm imputing all the correct values to the environment variables (using a bucket specific Key ID and App Key) but nothing appears to be working. Any suggestions?
